### DISCUSSION TITLE
How to access my local Database of the app?

### DISCUSSION THREAD

**Discussion by salexandru-123** [inventree/InvenTree#8453] [Q&A] [answered]
Good Evening,
I've been trying to access my local DB, using my localhost ip address and the DB_Port written in the .env file, how can I manage to access it? 
I've tried connecting using MySQL Workbench 8.0, it tells me that the connection to database failed. I've also added the username and password written in the .env file but nothing.

Please help.
Thanks in regards!

**Comment by SchrodingersGat** [comment]
Please provide more information about your setup:

- Are you runing via docker, the installer, or bare metal?
- What database backend have you configured?
- Are you trying to access the DB from the same computer that the server is runnin on?

**Reply by salexandru-123** [reply]
I'm running it via docker.
In the .env file I've configured it with postresql.
I'm trying to access it within the same computer.
I want to understand the app's database structure to migrate a big amount of data to it, I don't know if you have any method already done for migration and/or how it works.
Let me know, thanks!

**Reply by salexandru-123** [reply]
Should I try and run it bare metal instead of docker?

**Reply by SchrodingersGat** [reply]
> In the .env file I've configured it with postresql.

So, won't this cause an issue trying to connect via MySQL workbench?

**Reply by salexandru-123** [reply]
Indeed, I have to configure it with mysql instead? 
Sorry for my ignorance, but I'm a beginner developer.. 😃

**Reply by SchrodingersGat** [reply]
Either that, or use a dashboard tool intended for postgres

**Reply by salexandru-123** [reply]
I've tried using DBeaver today, it didn't work at all. I tried to connect in these ways.
First attempt:
- main host: {Docker container ip address}: 5432 (Docker container port set in the .yml file)
- superuser username and passoword
+
- ssh tunnel data:
   - DB_HOST:DB_PORT 
   - DB_NAME and DB_PASSWORD
This method didn't work at all.

Second attempt:
-I've tried to just use localhost:5432 and then my DB credentials and 
didn't work, also tried with my superuser credentials and still didn't work.
-I've also tried 127.0.0.1 instead of localhost.

**Reply by SchrodingersGat** [reply, marked as answer]
Right, so by default, the database container port (5432) is only exposed to the other docker containers.

To expose the port to the local machine, you will need to add the following line to your docker compose file:

```yaml
    ports:
      - 5432:5432
 ```

e.g.

![image](https://github.com/user-attachments/assets/841d31bd-345f-4ae4-bf4e-4c961e493aa9)

I have just tested this and confirmed that (with the additional line) it works with DBeaver

**Reply by salexandru-123** [reply]
Thanks it finally worked <3
